From bf4ed6e1ed9f3b29c9a7208e37a332aebe6b5ca5 Mon Sep 17 00:00:00 2001
From: Gerald Combs <gerald@wireshark.org>
Date: Mon, 26 Jan 2015 11:38:33 -0800
Subject: [PATCH 2/6] Pcapng: Don't fetch past the end of a GArray.

Due to an off-by-one error an invalid ISB interface ID could make us
fetch past the end of a GArray. Found using American Fuzzy Lop.

Bug: 10895
Change-Id: I7d4049ad7a386ae7e8013b8e741d54a31f353f1f
Reviewed-on: https://code.wireshark.org/review/6798
Petri-Dish: Gerald Combs <gerald@wireshark.org>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Gerald Combs <gerald@wireshark.org>
(cherry picked from commit a835c85e3d662343d7283f1dcdacb8a11d1d0727)
Conflicts: wiretap/pcapng.c
Reviewed-on: https://code.wireshark.org/review/6801
Reviewed-on: https://code.wireshark.org/review/7704
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
Tested-by: Balint Reczey <balint@balintreczey.hu>
---
 wiretap/pcapng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wiretap/pcapng.c b/wiretap/pcapng.c
index 45f1ad3..bb40dca 100644
--- a/wiretap/pcapng.c
+++ b/wiretap/pcapng.c
@@ -2385,7 +2385,7 @@ pcapng_read(wtap *wth, int *err, gchar **err_info, gint64 *data_offset)
                         pcapng_debug0("pcapng_read: block type BLOCK_TYPE_ISB");
                         *data_offset += bytes_read;
                         pcapng_debug1("pcapng_read: *data_offset is updated to %" G_GINT64_MODIFIER "d", *data_offset);
-                        if (wth->interface_data->len < wblock.data.if_stats.interface_id) {
+                        if (wth->interface_data->len <= wblock.data.if_stats.interface_id) {
                                 pcapng_debug1("pcapng_read: BLOCK_TYPE_ISB wblock.if_stats.interface_id %u > number_of_interfaces", wblock.data.if_stats.interface_id);
                         } else {
                                 /* Get the interface description */
-- 
2.1.4

